mesh/src/vertex/Cylindrical2dVertexMesh.cpp

```diff
@@ -150,40 +152,100 @@ Cylindrical2dVertexMesh::Cylindrical2dVertexMesh(Cylindrical2dMesh& rMesh, bool
+                     *   /\  
+                     *  /__\ edge of mesh 
+                     * 
+                     * Changing this can cause issues with stitching the left and right back together.
                      */
-                    unsigned num_sections = 2;
-                    for (unsigned section=0; section<=num_sections; section++)
-                    {
-                        double ratio = ((double)section)/(double)num_sections;
-                        c_vector<double,2> new_node_location = normal_vector + p_node_a->rGetLocation() + ratio*edge;
-
-                        //Check if near other nodes (could be inefficient)
-                        bool node_clear = true;
-                        double node_clearance = 0.05;
-
-                        for (unsigned i=0; i<nodes.size(); i++)
-                        {
-                            double distance = norm_2(mpDelaunayMesh->GetVectorFromAtoB(nodes[i]->rGetLocation(), new_node_location));
-                            if (distance < node_clearance)
-                            {
-                                node_clear = false;
-                                //break;
-                            }
+                    
+                    double ratio = 0.5;
+                    c_vector<double,2> new_node_location = normal_vector + p_node_a->rGetLocation() + ratio*edge;
+                    
+                    // Check if near other nodes (TODO look at this as could be ineficient)
+                    bool node_clear = true;
+                    double node_clearance = 0.01; 
+
+                    for (unsigned i=0; i<nodes.size(); i++)
+                    {   
+                        double distance = norm_2(mpDelaunayMesh->GetVectorFromAtoB(nodes[i]->rGetLocation(), new_node_location));
+                        if (distance < node_clearance)
+                        {   
+                            node_clear = false;
+                            //break;
                         }
+                    }
+                
+                    if (node_clear)
+                    {
+                        nodes.push_back(new Node<2>(new_node_index, new_node_location));
+                        new_node_index++;
+                    }
+                }
+            }
+        }
 
-                        if (node_clear)
-                        {
-                            nodes.push_back(new Node<2>(new_node_index, new_node_location));
-                            new_node_index++;
-                        }
+        // Add new nodes for voids (note these nodes won't be labeled as boundary nodes here)
+        for (TetrahedralMesh<2,2>::ElementIterator elem_iter = mpDelaunayMesh->GetElementIteratorBegin();
+            elem_iter != mpDelaunayMesh->GetElementIteratorEnd();
+            ++elem_iter)
+        {   
+            bool bad_element = false;
+            double edge_threshold = 1.5; //TODO Make setable variable!
```

Comment:
Agreed, this looks like it should be settable!

Reply:
I have reworded this and another todo and left as a future change as these chosen varies are sufficient for the current use cases.